packages/common/src/intl/utils/DateUtils.ts

```diff
@@ -0,0 +1,69 @@
+import { IntlUtils } from '@common/intl';
+import {
+  isValid,
+  differenceInMonths,
+  isPast,
+  isThisWeek,
+  isToday,
+  isThisMonth,
+  isAfter,
+  isBefore,
+  isEqual,
+  format,
+  parseISO,
+  fromUnixTime,
+} from 'date-fns';
+import { enGB, enUS, fr, ar } from 'date-fns/locale';
+
+// Map locale string (from i18n) to locale object (from date-fns)
+const getLocaleObj = { fr, ar };
+
+export const MINIMUM_EXPIRY_MONTHS = 3;
+
+const dateInputHandler = (date: Date | string | number): Date => {
+  // Assume a string is an ISO date-time string
+  if (typeof date === 'string') return parseISO(date);
+  // Assume a number is a UNIX timestamp
+  if (typeof date === 'number') return fromUnixTime(date);
+  return date as Date;
+};
+
+export const DateUtils = {
+  getDateOrNull: (date: string | null): Date | null => {
+    if (!date) return null;
+    const maybeDate = new Date(date);
+    return isValid(maybeDate) ? maybeDate : null;
+  },
+  isExpired: (expiryDate: Date): boolean => isPast(expiryDate),
+  isAlmostExpired: (
+    expiryDate: Date,
+    threshold = MINIMUM_EXPIRY_MONTHS
+  ): boolean => differenceInMonths(expiryDate, Date.now()) <= threshold,
+  isThisWeek,
+  isToday,
+  isThisMonth,
+  isAfter,
+  isBefore,
+  isEqual,
+};
+
+export const useFormatDateTime = () => {
+  const language = IntlUtils.useCurrentLanguage();
+  const locale =
+    language === 'en'
+      ? navigator.language === 'en-US'
+        ? enUS
+        : enGB
+      : getLocaleObj[language];
+
+  const localisedDate = (date: Date | string | number): string =>
+    format(dateInputHandler(date), 'P', { locale });
+
+  const customDate = (
+    date: Date | string | number,
+    formatString: string
+  ): string => format(dateInputHandler(date), formatString, { locale });
```

Comment:
Strictly we can do all date formatting with the one method (customDate), but useful to create a handful of "preset" formats (added as required)

Reply:
yes, the 'preset' idea is important. Good to have something like the expiry date formatted consistently throughout, and easy to update if required